activemodel/lib/active_model/type/helpers/numeric.rb

```diff
@@ -8,14 +8,21 @@ def serialize(value)
           cast(value)
         end
 
+        def serialize2(value)
```

Comment:
Why `serialize2` exists? All definitions of it in this PR seems to have the same code as `serialize`.

Reply:
It is used here https://github.com/rails/rails/pull/39009/files#diff-3bcf02e80bc52706c3df424d2a8c4db2R31 and it's a terrible name but we're not sure what to  call it. The other serialize methods do both serialize and validation, this one just does serialization.

Reply:
But if all the implementations of `serialize2` on this PR is always the same as the implementation as `serialize` I don't see the different and `serialize2` is also doing serialization and validation, no?

Reply:
It's not the same.  Check the serialize method on Integer [here](https://github.com/rails/rails/blob/2a12b723bbc38be74275910204d29b534e80f50c/activemodel/lib/active_model/type/integer.rb#L26-L29).  It's doing both serialization and validation.  This one is only doing serialization.

Reply:
Is the role of `serialize2` to be a fast, unchecked `serialize` (i.e. "garbage in, garbage out")?  If so, would `unchecked_serialize` be an appropriate name?

Reply:
Sorry, that was the context I was missing. In this diff all `serialize2` methods have the same implementation of the `serialize` in the same files, but I was missing that the subclasses were overriding the method.

If we want to `serialize2` to be part of the public API I'd go with @jonathanhefner's suggestion, but with the long term goal of renaming it to `serialize` and renaming the current `serialize` to something as `verify_and_serialize`.

Reply:
@rafaelfranca I like @jonathanhefner's suggestion too.  @eileencodes wdyt?

Reply:
Works for me!